docs/modules/nifi/pages/usage_guide/exposing-processors/http.adoc

```diff
@@ -0,0 +1,99 @@
+   This guide assumes you are receiving JSON documents, but you can also receive other files types, please update the record reader and writer accordingly.
+
+You should end up with something similar to
+
+image:listen-http-1.png[A ListenHTTP processor]
+
+== 2. Expose `ListenHTTP` processor
+
+Afterwards you need to expose the processor the the outside world of NiFi.
+
+For that to work, first create a Service object as follows.
+This guide assumes your NifiCluster is called `simple-nifi`, please update the YAML accordingly.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: simple-nifi-listen-http # Update according to NifiCluster name
+spec:
+  type: ClusterIP
+  selector:
+    app.kubernetes.io/component: node
+    app.kubernetes.io/instance: simple-nifi # Update according to NifiCluster name
+    app.kubernetes.io/name: nifi
+  ports:
+  - name: http
+    port: 8042
+    protocol: TCP
+    targetPort: 8042
+----
+
+In case you don't have an ingress controller, you can set the Service type to `LoadBalancer` or `NodePort` instead and should be ready to go.
+If you are using an ingress controller, an Ingress could look something like
+
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: simple-nifi-listen-http # Update according to NifiCluster name
+spec:
+  rules:
+  - host: simple-nifi-listen-http.my.corp # Update to your host
+    http:
+      paths:
+      - pathType: Prefix
+        path: /
+        backend:
+          service:
+            name: simple-nifi-listen-http # Update to your Service name
+            port:
+              number: 8042
+----
+
+=== 3. Route 
+
+First, create a `RouteOnAttribute` processor and connect it to the `success` output of the `ListenHTTP` processor as shown below.
+Start the `ListenHTTP` processor.
+
+image:listen-http-2.png[´RouteOnAttribute connected to ListenHTTP processor]
+
+The `ListenHTTP` processor should now generate a FlowFile for every incoming HTTP request.
+You can test this e.g. by calling `curl --verbose --data '{"hello":"NiFi"}' https://simple-nifi-listen-http.my.corp`, you should get a `HTTP/2 200` response.
+
+TIP: If you get a `503 Service Temporarily Unavailable`, this probably means your ingress controller was not able to
+reach your `ListenHTTP` processor, it is probably not running or not configured correctly.
+
+This should result in one FlowFile being queued, as shown in the picture above and below.
+
+image:listen-http-3.png[Resulting FlowFile]
+
+The next step is to handle different kind of messages coming in, based on the HTTP path.
+
+In the `RouteOnAttribute` processor, add a field called `/webhook/foo` with the value of `${"restlistener.request.uri":equals('/webhook/foo')}`.
+You can replace `/webhook/foo` with whatever URL your HTTP service should be reachable.
+This guide also added `/webhook/bar` and `/webhook/bak` in a similar way.
```

Comment:
I didn't see where `bar` and `bak` were used.

Also, in case it is a typo:

```suggestion
This guide also added `/webhook/bar` and `/webhook/baz` in a similar way.
```

Reply:
They are in the screenshot docs/modules/nifi/assets/images/listen-http-4.png, I wanted to showcase routing to more than one target ^^